logger.go

```diff
@@ -250,6 +250,21 @@ func (log *Logger) Core() zapcore.Core {
 	return log.core
 }
 
+func (log *Logger) SetCore(core zapcore.Core) {
+	log.core = core
+}
+
+func (log *Logger) Clone() *Logger {
+	return log.clone()
+}
+
+func (log *Logger) CloneWithNewCore(core zapcore.Core) *Logger {
```

Comment:
You can use `logger.WithOptions(zap.WrapCore(...))` to create a new logger with the core wrapped or replaced completely